client/src/app/shared/models/poll/view-poll.ts

```diff
@@ -0,0 +1,141 @@
+import { ViewAssignment } from 'app/site/assignments/models/view-assignment';
+import { BaseProjectableViewModel } from 'app/site/base/base-projectable-view-model';
+import { BaseViewModel } from 'app/site/base/base-view-model';
+import { ProjectorElementBuildDeskriptor } from 'app/site/base/projectable';
+import { HasMeeting } from 'app/site/event-management/models/view-meeting';
+import { ViewMotion } from 'app/site/motions/models/view-motion';
+import { ViewGroup } from 'app/site/users/models/view-group';
+import { ViewUser } from 'app/site/users/models/view-user';
+import { Poll } from './poll';
+import {
+    MajorityMethodVerbose,
+    PollClassType,
+    PollClassTypeVerbose,
+    PollMethodVerbose,
+    PollPercentBaseVerbose,
+    PollStateChangeActionVerbose,
+    PollStateVerbose,
+    PollType,
+    PollTypeVerbose
+} from './poll-constants';
+import { ViewOption } from './view-option';
+
+export class ViewPoll extends BaseProjectableViewModel<Poll> {
+    public get poll(): Poll {
+        return this._model;
+    }
+    public static COLLECTION = Poll.COLLECTION;
+    protected _collection = Poll.COLLECTION;
+
+    public get pollClassType(): PollClassType {
+        return this.content_object?.collection as PollClassType;
+    }
+
+    public get isAssignmentPoll(): boolean {
+        return this.content_object?.collection === PollClassType.Assignment;
+    }
+
+    public get isMotionPoll(): boolean {
+        return this.content_object?.collection === PollClassType.Motion;
+    }
+
+    public get pollmethodVerbose(): string {
+        return PollMethodVerbose[this.pollmethod];
+    }
+
+    public get percentBaseVerbose(): string {
+        return PollPercentBaseVerbose[this.onehundred_percent_base];
+    }
+
+    public get pollClassTypeVerbose(): string {
+        return PollClassTypeVerbose[this.pollClassType];
+    }
+
+    public get parentLink(): string {
+        const routeFragments = this.content_object_id.split('/');
+        return `/${routeFragments[0]}s/${routeFragments[1]}`;
+    }
+
+    public get stateVerbose(): string {
+        return PollStateVerbose[this.state];
+    }
+
+    public get nextStateActionVerbose(): string {
+        return PollStateChangeActionVerbose[this.nextState];
+    }
+
+    public get typeVerbose(): string {
+        return PollTypeVerbose[this.type];
+    }
+
+    public get majorityMethodVerbose(): string {
+        return MajorityMethodVerbose[this.majority_method];
+    }
+
+    public get hasVote(): boolean | null {
+        if (!this.user_has_voted && this.canBeVotedFor()) {
+            return false;
+        } else if (this.user_has_voted) {
+            return true;
+        } else {
+            return null;
```

Comment:
Add docs about the meaning of `null`. I think you always should return a boolean.